### ISSUE TITLE
Feature 8.6 - MiniBatchSettings dialog

### ISSUE THREAD

**Issue by borislav-vucicevic-04** [borislav-vucicevic-04/regressly#43]
Due to the recent change of plans, the application should run mini-batch gradient descent and not gradient descent. This means we need to provide the ability to users to choose the following:
- Number of epochs
- Batch size
- Learning rate

So implement a new component that will be a modal dialog window where users can enter those values.

The dialog must ensure that the user entered valid values in textboxes; otherwise, it will throw the error. Users will be forced to enter valid values unless they click cancel.

Valid values are:
- For the **number of epochs**: positive integer values
- For the **batch size**: positive integer values smaller than the number of entries in the dataset.
- For the **learning rate**: positive floating-point values

**Comment by borislav-vucicevic-04** [comment]
Due to recent changes in plans, we are not implementing mini-batch gradient descent and this component will be deleted
